dive_core/command_hierarchy.cpp

```diff
@@ -2089,32 +2093,157 @@ void CommandHierarchyCreator::AppendPacketFieldNodes(const IMemoryManager &mem_m
+                field_string_stream << prefix << "(DWORD " << i << "): 0x" << std::hex
+                                    << dword_value;
+
+                CommandHierarchy::AuxInfo aux_info = CommandHierarchy::AuxInfo::RegFieldNode(false);
+                uint64_t                  field_node_index = AddNode(NodeType::kFieldNode,
+                                                    field_string_stream.str(),
+                                                    aux_info);
+
+                // Add it as child to packet_node
+                AddChild(CommandHierarchy::kEngineTopology, packet_node_index, field_node_index);
+                AddChild(CommandHierarchy::kSubmitTopology, packet_node_index, field_node_index);
+                AddChild(CommandHierarchy::kAllEventTopology, packet_node_index, field_node_index);
+                AddChild(CommandHierarchy::kRgpTopology, packet_node_index, field_node_index);
+            }
+        }
+    }
+}
 
-            CommandHierarchy::AuxInfo aux_info = CommandHierarchy::AuxInfo::RegFieldNode(
-            is_ce_packet);
-            uint64_t field_node_index = AddNode(NodeType::kFieldNode,
-                                                field_string_stream.str(),
-                                                aux_info);
+//--------------------------------------------------------------------------------------------------
+void CommandHierarchyCreator::AppendLoadStateExtBufferNode(const IMemoryManager &mem_manager,
+                                                           uint32_t              submit_index,
+                                                           uint64_t              va_addr,
+                                                           uint64_t              packet_node_index)
+{
+    PM4_CP_LOAD_STATE6 packet;
+    DIVE_VERIFY(mem_manager.CopyMemory(&packet, submit_index, va_addr, sizeof(packet)));
 
-            // Add it as child to packet_node
-            AddChild(CommandHierarchy::kEngineTopology, packet_node_index, field_node_index);
-            AddChild(CommandHierarchy::kSubmitTopology, packet_node_index, field_node_index);
-            AddChild(CommandHierarchy::kAllEventTopology, packet_node_index, field_node_index);
-            AddChild(CommandHierarchy::kRgpTopology, packet_node_index, field_node_index);
+    enum class StateBlockCat
+    {
+        kTex,
+        kShader,
+        kIbo
+    };
+    StateBlockCat cat;
+    switch (packet.bitfields0.STATE_BLOCK)
+    {
+    case SB6_VS_TEX:
+    case SB6_HS_TEX:
+    case SB6_DS_TEX:
+    case SB6_GS_TEX:
+    case SB6_FS_TEX:
+    case SB6_CS_TEX: cat = StateBlockCat::kTex; break;
+    case SB6_VS_SHADER:
+    case SB6_HS_SHADER:
+    case SB6_DS_SHADER:
+    case SB6_GS_SHADER:
+    case SB6_FS_SHADER:
+    case SB6_CS_SHADER: cat = StateBlockCat::kShader; break;
+    case SB6_IBO:
+    case SB6_CS_IBO:
+        DIVE_ASSERT("These seem to be specific to pre-A6XX");
```

Comment:
oh, i didn't know that we are allowed to use DIVE_ASSERT this way. did a quick search, this seems to be the only place that use it as warning message?

Reply:
What do you mean by "this way"? I'm adding an assert there because I don't ever expect a command buffer with these parameters. But I still want to catch them just-in-case. Do you prefer I use something else instead?

Reply:
so should be `DIVE_ASSERT(false, "These seem to be specific to pre-A6XX");`?

Reply:
Oh oops, you're right!